src/components/Api/DataFetcher.tsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+import useSWRImmutable from 'swr/immutable';
+
+import ErrorMessage from './ErrorMessage';
+
+import { fetcher, OFFLINE_ERROR } from 'src/api';
+import Spinner from 'src/components/dls/Spinner/Spinner';
+import { BaseResponse } from 'types/ApiResponses';
+
+interface Props {
+  queryKey: string;
+  render: (data: BaseResponse) => JSX.Element;
+  initialData?: BaseResponse;
+}
+
+const DataFetcher: React.FC<Props> = ({ queryKey, render, initialData }) => {
```

Comment:
2 comments. 

1) We should add some documentation to explain why we need the datafetcher component and when to use it. 
2) nit: I prefer components/DataFetcher over /api, sounds counter-intuitive to have a component directory for the API